mirdata/datasets/tonas.py

```diff
@@ -0,0 +1,409 @@
+        singer (str): performing singer (cantaor)
+        title (str): title of the track song
+        tuning_frequency (float): tuning frequency of the symbolic notation
+
+    Cached Properties:
+        melody (F0Data): annotated melody in extended F0Data format
+        notes (NoteData): annotated notes
+
+    """
+
+    def __init__(
+        self,
+        track_id,
+        data_home,
+        dataset_name,
+        index,
+        metadata,
+    ):
+        super().__init__(
+            track_id,
+            data_home,
+            dataset_name,
+            index,
+            metadata,
+        )
+
+        self.f0_path = self.get_path("f0")
+        self.notes_path = self.get_path("notes")
+
+        self.audio_path = self.get_path("audio")
+
+    @property
+    def style(self):
+        return self._track_metadata.get("style")
+
+    @property
+    def singer(self):
+        return self._track_metadata.get("singer")
+
+    @property
+    def title(self):
+        return self._track_metadata.get("title")
+
+    @property
+    def tuning_frequency(self):
+        return load_tuning_frequency(self.notes_path)
+
+    @property
+    def audio(self) -> Tuple[np.ndarray, float]:
+        """The track's audio
+
+        Returns:
+            * np.ndarray - audio signal
+            * float - sample rate
+
+        """
+        return load_audio(self.audio_path)
+
+    @core.cached_property
+    def f0(self) -> Optional[F0Data]:
+        return load_f0(self.f0_path)
+
+    @core.cached_property
+    def notes(self) -> Optional[NoteData]:
+        return load_notes(self.notes_path)
+
+    def to_jams(self):
+        """Get the track's data in jams format
+
+        Returns:
+            jams.JAMS: the track's data in jams format
+
+        """
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path,
+            f0_data=[(self.f0, "pitch_contour")],
+            note_data=[(self.notes, "note_hz")],
+            metadata=self._track_metadata,
+        )
+
+
+def load_audio(fhandle: str) -> Tuple[np.ndarray, float]:
+    """Load a TONAS audio file.
+
+    Args:
+        fhandle (str): path to an audio file
+
+    Returns:
+        * np.ndarray - the mono audio signal
+        * float - The sample rate of the audio file
+
+    """
+    return librosa.load(fhandle, sr=44100, mono=True)
+
+
+@io.coerce_to_string_io
+def load_f0(fhandle: TextIO) -> Optional[F0Data]:
+    """Load TONAS f0 annotations
+
+    Args:
+        fhandle (str or file-like): path or file-like object pointing to f0 annotation file
+
+    Returns:
+        F0Data: predominant f0 melody
+
+    """
+    times = []
+    freqs = []
+    freqs_corr = []
+    energies = []
+    reader = np.genfromtxt(fhandle)
+    for line in reader:
+        times.append(float(line[0]))
+        energies.append(float(line[1]))
+        freqs.append(float(line[2]))
+        freqs_corr.append(float(line[3]))
+
+    times = np.array(times, dtype="float")  # type: ignore
```

Comment:
What do you mean by `# type: ignore`?

Reply:
Nothing sorry! Copied, pasted, and adapted the function from another loader, and forgot to remove these comments!